autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -0,0 +1,180 @@
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private LoadAlertRepository alertRepository;
+
+    @Inject
+    private EventPublisher eventPublisher;
+
+    @Inject
+    private YarnMetricsClient yarnMetricsClient;
+
+    @Inject
+    private StackResponseUtils stackResponseUtils;
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    private long clusterId;
+
+    private Cluster cluster;
+
+    private LoadAlert loadAlert;
+
+    @Nonnull
+    @Override
+    public EvaluatorContext getContext() {
+        return new ClusterIdEvaluatorContext(clusterId);
+    }
+
+    @Override
+    public void setContext(EvaluatorContext context) {
+        clusterId = (long) context.getData();
+    }
+
+    @Override
+    public String getName() {
+        return EVALUATOR_NAME;
+    }
+
+    @Override
+    protected void execute() {
+        cluster = clusterService.findById(clusterId);
+        loadAlert = cluster.getLoadAlerts().stream().findFirst().get();
+
+        long remainingTime = ClusterUtils.getRemainingCooldownTime(
+                loadAlert.getLoadAlertConfiguration().getCoolDownMillis(),
+                cluster.getLastScalingActivity());
+
+        if (remainingTime <= 0) {
+            pollYarnMetricsAndScaleCluster();
+        } else {
+            LOGGER.debug("Cluster cannot be scaled for {} min(s)",
+                    ClusterUtils.TIME_FORMAT.format((double) remainingTime / TimeUtil.MIN_IN_MS));
+        }
+    }
+
+    protected void pollYarnMetricsAndScaleCluster() {
+        long start = System.currentTimeMillis();
+        try {
+            MDCBuilder.buildMdcContext(cluster);
+
+            StackV4Response stackV4Response = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
+
+            String hostGroupInstanceType =
+                    stackResponseUtils.getHostGroupInstanceType(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
+
+            YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
+                    .getYarnMetricsForCluster(cluster, hostGroupInstanceType, stackV4Response.getCloudPlatform());
+
+            Map<String, String> hostFqdnsToInstanceId = stackResponseUtils
+                    .getCloudInstanceIdsForHostGroup(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
+            yarnResponse.getScaleUpCandidates().ifPresentOrElse(
+                    scaleUpCandidates -> handleScaleUp(hostGroupInstanceType, scaleUpCandidates, hostFqdnsToInstanceId.size()),
+                    () -> {
+                        yarnResponse.getScaleDownCandidates().ifPresent(
+                                scaleDownCandidates -> handleScaleDown(scaleDownCandidates, hostFqdnsToInstanceId));
+                    });
+
+            LOGGER.debug("Finished loadEvaluator for cluster {} in {} ms", cluster.getStackCrn(), System.currentTimeMillis() - start);
```

Comment:
From testing, this is fairly quick, correct?

Reply:
local testing it is pretty quick.